src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -517,9 +516,18 @@ def from_pandas(
                 ordered_dataframe=ordered_dataframe,
                 data_column_pandas_labels=original_column_labels,
                 data_column_pandas_index_names=original_column_index_names,
+                data_column_types=snowpark_types[
+                    len(index_snowflake_quoted_identifiers) : (
```

Comment:
We could change `infer_snowpark_types_from_pandas` to return `snowpark_pandas_data_column_types`, `snowpark_pandas_index_column_types`, and `snowpark_types` and directly use them here.

Reply:
That doesn't make sense to me because `infer_snowpark_types_from_pandas` shouldn't have the context of how the pandas dataframe it gets was constructed (specifically, that it contains the original index columns, followed by data columns, then a row position column), whereas `from_pandas` should have that context. I prefer the current version.

Reply:
I see. I was confused because I didn't read the code above which preprocessed `df` to be the original df plus the index.